### PR TITLE
Interaction - Battlefields: Cloister Areas

### LINKED ISSUE #5777: 🔨 Request: Quest Progress check for new Battlefield framework.
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my issue will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.

## Describe the feature

There are some quests that have battlefields associated with them but I don't think we currently have a check for quest progression in the newer battlefield system unless I'm missing/overlooked something.

### LINKED ISSUE #5783: 🐛 Error on startup in scripts/globals/interaction/interaction_global.lua
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my issue will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.

## OS / platform the server is running (if known)

Server 2019 Std

## Branch affected by issue 

`base`

## Steps to reproduce

Start the server

## Expected behavior

I expect it to load without error.

### Error:
[05/18/24 13:02:50:341][map][error] luautils::InitInteractionGlobal: ./scripts/globals/interaction/interaction_global.lua:46: attempt to index a boolean value
stack traceback:
	./scripts/globals/interaction/interaction_global.lua:46: in function 'loadContainers'
	./scripts/globals/interaction/interaction_global.lua:26: in function <./scripts/globals/interaction/interaction_global.lua:16> (luautils::InitInteractionGlobal:1151)

### PR DESCRIPTION
**Pull request by claywar**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
* Converts all existing battlefields in Cloister areas to Interaction Framework
* Adds requiredVar/requiredValue options to BattlefieldQuest to support checking quest progress (Closes #5777)
* Moves bcnm table of battlefield indices and ids to documentation, and removes lua references for completed zones

## Steps to test these changes
Enter/win/exit/lose the converted BCNMs

### CODE REVIEW

**Review by Xaver-DaRed — DISMISSED** [review]

File: scripts/battlefields/Cloister_of_Gales/carbuncle_debacle.lua [hunk 1]
```diff
@@ -0,0 +1,48 @@
+-----------------------------------
+-- Area: Cloister of Gales
+-- BCNM: Carbuncle Debacle
+-----------------------------------
+local cloisterOfGalesID = zones[xi.zone.CLOISTER_OF_GALES]
+-----------------------------------
+
+local content = BattlefieldQuest:new({
+    zoneId           = xi.zone.CLOISTER_OF_STORMS,
```
Comment (Frankie-hz) [inline scripts/battlefields/Cloister_of_Gales/carbuncle_debacle.lua]: zoneId should be GALES

File: scripts/battlefields/Cloister_of_Tremors/trial-size_trial_by_earth.lua [hunk 2]
```diff
@@ -0,0 +1,44 @@
+-----------------------------------
+-- Area: Cloister of Flames
```
Comment (Frankie-hz) [inline scripts/battlefields/Cloister_of_Tremors/trial-size_trial_by_earth.lua]: Cloiser of Tremors

File: scripts/battlefields/Cloister_of_Frost/class_reunion.lua [hunk 3]
```diff
@@ -0,0 +1,69 @@
+-----------------------------------
+-- Area: Cloister of Tremors
```
Comment (Frankie-hz) [inline scripts/battlefields/Cloister_of_Frost/class_reunion.lua]: Cloister of Frost

**Review by Xaver-DaRed — APPROVED** [review]

File: scripts/battlefields/Cloister_of_Tremors/trial_by_earth.lua [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    timeLimit        = utils.minutes(30),
+    index            = 0,
+    entryNpc         = 'EP_Entrance',
+    exitNpc          = 'Earth_Protocrystal',
+    requiredKeyItems = { xi.ki.TUNING_FORK_OF_EARTH },
+
+    questArea = xi.questLog.BASTOK,
+    quest     = xi.quest.id.outlands.TRIAL_BY_EARTH,
+})
+
```
Comment (Frankie-hz) [inline scripts/battlefields/Cloister_of_Tremors/trial_by_earth.lua:21]: I think this is missing a function:  `function content:onEventFinishWin(player, csid, option, npc` for the title and keyitem

**Review by Frankie-hz — CHANGES_REQUESTED** [review]

File: scripts/battlefields/Cloister_of_Tremors/trial_by_earth.lua (hunk 4 above)
Comment (claywar) [reply scripts/battlefields/Cloister_of_Tremors/trial_by_earth.lua:21]: Its not.  That's the only Trial covered by interaction for the quest

Comment (Frankie-hz) [reply scripts/battlefields/Cloister_of_Tremors/trial_by_earth.lua:21]: Ahh ok, I did not realize that. Looks good then! Huge thanks for doing this 😄

**Pull request merged**